src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -193,7 +198,19 @@ public static int Main(string[] args)
             return result.ExitCode == 0;
         }
 
-        private static bool Compile(ProjectContext context, string configuration, string outputOptionValue, string intermediateOutputValue, bool buildProjectReferences, bool noHost)
+        private static Command ResolveCommand(string commandName, string args, string projectName, Preconditions preconditions)
```

Comment:
This function doesn't seem to be resolving anything. `DetermineCommandSafety` or something similar seems like a more appropriate name here. 

I would actually prefer to see this return something rather than modifying this Preconditions object everywhere.